src/App.js

```diff
@@ -554,127 +555,123 @@ export class App extends React.Component<AppProps, AppState> {
-                                        interpreterIsRunning={this.state.runningState === 'running'}
-                                        disabled={this.state.programSequence.getProgramLength() === 0}
-                                        onClick={this.handleClickPlay}
+                    }
+                    */}
+                    <div className='App__command-palette'>
+                        <h2 className='App__command-palette-heading App__commandpalette-heading-long'>
+                            <FormattedMessage id='CommandPalette.movementsTitle' />
+                        </h2>
+                        <h2 className='App__command-palette-heading App__commandpalette-heading-short'>
+                            <FormattedMessage id='CommandPalette.shortMovementsTitle' />
+                        </h2>
+                        <div className='App__command-palette-command-container'>
+                            <div className='App__command-palette-commands'>
+                                {this.renderCommandBlocks()}
+                            </div>
+                        </div>
+                    </div>
+                    <div className='App__scene-container'>
+                        <Scene
+                            dimensions={this.state.sceneDimensions}
+                            characterState={this.state.characterState}
+                            theme={this.state.settings.theme}
+                        />
+                        <div className='App__scene-controls'>
+                            <div className='App__scene-controls-group'>
+                                <PenDownToggleSwitch
+                                    className='App__penDown-toggle-switch'
+                                    value={this.state.drawingEnabled}
+                                    onChange={this.handleTogglePenDown}/>
+                                <div className='App__refreshButton-container'>
+                                    <RefreshButton
+                                        disabled={this.state.runningState === 'running'}
+                                        onClick={this.handleRefresh}
                                     />
-                                    <StopButton
-                                        disabled={
-                                            this.state.runningState === 'stopped'
-                                            || this.state.runningState === 'stopRequested'}
-                                        onClick={this.handleClickStop}/>
                                 </div>
+                            </div>
+                        </div>
+                    </div>
+                    <div className='App__program-block-editor'>
+                        <ProgramBlockEditor
+                            actionPanelStepIndex={this.state.actionPanelStepIndex}
+                            editingDisabled={this.state.runningState === 'running'}
```

Comment:
It seems I have made a mistake, I will change it back to what it was before.

Reply:
Though, aren't we allowing editing when the program is paused?

Reply:
Sorry, I misread the condition.

Reply:
Yes, we can edit when paused and this expression allows it: editing is enabled when 'stopped' or 'paused'. It's a little hard to read because of the double negative: `editingDisabled = !(...)`.

If `runningState` is `paused`, then:

- `this.state.runningState === 'paused'` is true
- `(this.state.runningState === 'stopped' || this.state.runningState === 'paused')` is true
- `!(this.state.runningState === 'stopped' || this.state.runningState === 'paused')` is false
- `editingDisabled` is set to false, so editing is enabled

Reply:
Thanks for catching this and your explanations, it makes sense to change this back to the previous condition, as the previous condition was handling cases for "stopRequested" and "pauseRequested".